### PR TITLE
Clean up IAM Scopes in the documentation

### PR DESCRIPTION
**Pull request by upodroid**
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by upodroid** [comment]
This issue should be closed https://github.com/hashicorp/terraform-provider-google/issues/1943

File: templates/terraform/examples/base_configs/test_file.go.erb [hunk 1]
```diff
@@ -10,8 +10,6 @@ import (
   "github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 <%
-# raise 'skip_import_test should be true if resource is not importable' \
```
Comment (upodroid) [inline templates/terraform/examples/base_configs/test_file.go.erb:13]: Follow up from https://github.com/GoogleCloudPlatform/magic-modules/pull/4273

**Comment by upodroid** [comment]
@rileykarson while i'm cleaning up legacy scopes, can i remove compute, devstorage and clouddns from 

https://github.com/hashicorp/terraform-provider-google/blob/179ede2188a1c16c3d17dd7e3099b616297000d8/google/config.go#L219

given that cloud-platform supercedes them?

File: third_party/terraform/website/docs/r/container_cluster.html.markdown [hunk 2]
```diff
@@ -59,54 +64,15 @@ resource "google_container_node_pool" "primary_preemptible_nodes" {
...
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
+    service_account = google_service_account.default.email
+    oauth_scopes    = [
       "https://www.googleapis.com/auth/cloud-platform"
     ]
   }
 }
 ```
 
-## Example Usage - with the default node pool
```
Comment (rileykarson) [inline third_party/terraform/website/docs/r/container_cluster.html.markdown:70]: drive-by: This + the node pool example being removed are intentional and should be kept, even if they don't fit the best practices for configuring the resource. While we'd prefer users configure resources using the recommended setups, I'd rather we provide clear documentation on what alternatives look like vs forcing users to experiment themselves.

**Review by rileykarson — COMMENTED** [review]
@upodroid: Can you file an issue? I don't want to change the declared list of scopes outside of a major version, particularly if we're removing any.

File: third_party/terraform/website/docs/r/container_cluster.html.markdown (hunk 2 above)
Comment (upodroid) [reply third_party/terraform/website/docs/r/container_cluster.html.markdown:70]: I'll revert this change for both resources.

Comment (upodroid) [reply third_party/terraform/website/docs/r/container_cluster.html.markdown]: The GKE resource is such a mess. There are so many open breaking change issues for it.

Comment (rileykarson) [reply third_party/terraform/website/docs/r/container_cluster.html.markdown]: It's a busy resource!

File: third_party/terraform/website/docs/r/composer_environment.html.markdown [hunk 3]
```diff
@@ -44,9 +45,8 @@ resource "google_composer_environment" "test" {
 
 ### With GKE and Compute Resource Dependencies
 
-**NOTE** To use service accounts, you need to give `role/composer.worker` to the service account on any resources that may be created for the environment
-(i.e. at a project level). This will probably require an explicit dependency
-on the IAM policy binding (see `google_project_iam_member` below).
+**NOTE** To use custom service accounts, you need to give at least `role/composer.worker` to the service account being used Composer Nodes on the Composer project.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/composer_environment.html.markdown:48]: It seems like there's something missing between "being used" and "Composer Nodes".

File: third_party/terraform/website/docs/r/composer_environment.html.markdown [hunk 4]
```diff
@@ -65,8 +65,6 @@ resource "google_composer_environment" "test" {
       service_account = google_service_account.test.name
     }
   }
-
-  depends_on = [google_project_iam_member.composer-worker]
```
Comment (melinath) [inline third_party/terraform/website/docs/r/composer_environment.html.markdown:69]: Why is the explicit dependency no longer required?

File: third_party/terraform/website/docs/r/composer_environment.html.markdown [hunk 5]
```diff
@@ -44,9 +45,8 @@ resource "google_composer_environment" "test" {
 
 ### With GKE and Compute Resource Dependencies
 
-**NOTE** To use service accounts, you need to give `role/composer.worker` to the service account on any resources that may be created for the environment
-(i.e. at a project level). This will probably require an explicit dependency
-on the IAM policy binding (see `google_project_iam_member` below).
+**NOTE** To use custom service accounts, you need to give at least `role/composer.worker` to the service account being used Composer Nodes on the Composer project.
+You may need to assign additional roles depending on what the Airflow DAGs will be running.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/composer_environment.html.markdown:49]: I'm not familiar with GKE resources, so I would just be confused by "Airflow DAGs" and "Composer Nodes" compared to the old docs. Is there a different way to phrase this? Or is it something where if I were doing this, I would absolutely know what these phrases refer to?

File: third_party/terraform/website/docs/r/compute_instance.html.markdown [hunk 6]
```diff
@@ -51,7 +56,9 @@ resource "google_compute_instance" "default" {
   metadata_startup_script = "echo hi > /test.txt"
 
   service_account {
-    scopes = ["userinfo-email", "compute-ro", "storage-ro"]
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/compute_instance.html.markdown:59]: I think this can be simplified to indicate the current state rather than a change.
```suggestion
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

Would it make sense / be possible to add a link to the section of the docs explaining this recommendation in more detail?

File: third_party/terraform/website/docs/r/compute_instance_template.html.markdown [hunk 7]
```diff
@@ -61,7 +66,9 @@ resource "google_compute_instance_template" "default" {
   }
 
   service_account {
-    scopes = ["userinfo-email", "compute-ro", "storage-ro"]
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/compute_instance_template.html.markdown:69]:
```suggestion
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

File: third_party/terraform/website/docs/r/container_cluster.html.markdown [hunk 8]
```diff
@@ -56,11 +52,9 @@ resource "google_container_node_pool" "primary_preemptible_nodes" {
     preemptible  = true
     machine_type = "e2-medium"
 
-    metadata = {
-      disable-legacy-endpoints = "true"
-    }
-
-    oauth_scopes = [
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/container_cluster.html.markdown:55]:
```suggestion
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

File: third_party/terraform/website/docs/r/container_cluster.html.markdown [hunk 9]
```diff
@@ -74,32 +68,17 @@ resource "google_container_cluster" "primary" {
...
-    username = ""
-    password = ""
-
-    client_certificate_config {
-      issue_client_certificate = false
-    }
-  }
-
   node_config {
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/container_cluster.html.markdown:72]:
```suggestion
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

File: third_party/terraform/website/docs/r/container_node_pool.html.markdown [hunk 10]
```diff
@@ -40,6 +45,8 @@ resource "google_container_node_pool" "primary_preemptible_nodes" {
     preemptible  = true
     machine_type = "e2-medium"
 
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/container_node_pool.html.markdown:48]:
```suggestion
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

File: third_party/terraform/website/docs/r/container_node_pool.html.markdown [hunk 11]
```diff
@@ -50,12 +57,23 @@ resource "google_container_node_pool" "primary_preemptible_nodes" {
...
+
 resource "google_container_node_pool" "np" {
   name       = "my-node-pool"
   location   = "us-central1-a"
   cluster    = google_container_cluster.primary.name
-  node_count = 3
-
+  node_config {
+    machine_type = "e2-medium"
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/container_node_pool.html.markdown:71]:
```suggestion
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

File: third_party/terraform/website/docs/r/container_node_pool.html.markdown [hunk 12]
```diff
@@ -71,24 +89,12 @@ resource "google_container_cluster" "primary" {
...
-    username = ""
-    password = ""
-
-    client_certificate_config {
-      issue_client_certificate = false
-    }
-  }
-
   node_config {
+    # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/container_node_pool.html.markdown:93]:
```suggestion
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

File: third_party/terraform/website/docs/r/dataproc_cluster.html.markdown [hunk 13]
```diff
@@ -73,11 +78,10 @@ resource "google_dataproc_cluster" "mycluster" {
 
     gce_cluster_config {
       tags = ["foo", "bar"]
+      # Google now recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (melinath) [inline third_party/terraform/website/docs/r/dataproc_cluster.html.markdown:81]:
```suggestion
      # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```

**Review by melinath — CHANGES_REQUESTED** [review]
LGTM for the most part - I have a few questions, and one suggested change to the new standard language explaining service account usage.

File: third_party/terraform/website/docs/r/composer_environment.html.markdown (hunk 4 above)
Comment (upodroid) [reply third_party/terraform/website/docs/r/composer_environment.html.markdown:69]: That dependency isn't required. It takes time for the GKE cluster to be provisioned and for Airflow to start making API calls with that service account. The IAM grant will be finished and consistent long before the cluster is ready.

File: third_party/terraform/website/docs/r/compute_instance.html.markdown (hunk 6 above)
Comment (upodroid) [reply third_party/terraform/website/docs/r/compute_instance.html.markdown:59]: I don't think so. I'm assuming terraform users know what custom service accounts.

File: third_party/terraform/website/docs/r/composer_environment.html.markdown (hunk 5 above)
Comment (upodroid) [reply third_party/terraform/website/docs/r/composer_environment.html.markdown:49]: For example, if you write a DAG that talks to Bigquery, you need to assign additional roles. roles/composer.worker doesn't include BQ roles.

https://cloud.google.com/composer/docs/how-to/using/writing-dags#gcp_operators

I clarified it a little bit.

**Review by melinath — APPROVED** [review]
lgtm - thanks!

**Comment by melinath** [comment]
@upodroid I ran into an issue where using the service account names in the new documentation causes an error because of the use of `_` instead of `-`. Is there any chance you would be able to put together a follow-up PR switching those over?

```
Error: "account_id" ("service_account_id") doesn't match regexp "^[a-z](?:[-a-z0-9]{4,28}[a-z0-9])$"
```

**Comment by upodroid** [comment]
Hmm, which example was that?

**Comment by upodroid** [comment]
nvm, underscores are illegal in service account names
